composer/loggers/mlflow_logger.py

```diff
@@ -139,8 +147,50 @@ def __init__(
                 else:
                     self._experiment_id = (self._mlflow_client.create_experiment(name=self.experiment_name))
 
-    def init(self, state: State, logger: Logger) -> None:
+    def _start_mlflow_run(self, state):
         import mlflow
+
+        env_run_id = os.getenv(
+            mlflow.environment_variables.MLFLOW_RUN_ID.name,  # pyright: ignore[reportGeneralTypeIssues]
+            None,
+        )
+        if env_run_id is not None:
+            self._run_id = env_run_id
+        elif self.resume:
+            # Search for an existing run tagged with this Composer run if `self.resume=True`.
+            assert self._experiment_id is not None
+            existing_runs = mlflow.search_runs(
+                experiment_ids=[self._experiment_id],
+                filter_string=f'tags.run_name = "{state.run_name}"',
+                output_format='list',
+            )
+
+            if len(existing_runs) > 0:
+                self._run_id = existing_runs[0].info.run_id
+            else:
+                new_run = self._mlflow_client.create_run(
+                    experiment_id=self._experiment_id,
+                    run_name=self.run_name,
+                )
+                self._run_id = new_run.info.run_id
```

Comment:
```suggestion
            else:
                log.debug('Creating a new mlflow run as resume was set to True but no previous run was found.')
                new_run = self._mlflow_client.create_run(
                    experiment_id=self._experiment_id,
                    run_name=self.run_name,
                )
                self._run_id = new_run.info.run_id
```

Reply:
This shouldnt be a warning because a user likely leaves this on (eg how we treat autoresume for checkpoints), but a debug warning will be helpful if it is unexpected